src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.Stream.cs

```diff
@@ -0,0 +1,347 @@
+    using System.Threading.Tasks;
+
+    internal sealed partial class ODataUtf8JsonWriter
+    {
+        /// <summary>
+        /// Starts a scope for writing a stream value.
+        /// </summary>
+        /// <returns>A stream for writing the stream value.</returns>
+        public Stream StartStreamValueScope()
+        {
+            this.CommitUtf8JsonWriterContentsToBuffer();
+
+            this.WriteItemWithSeparatorIfNeeded();
+
+            this.bufferWriter.Write(this.DoubleQuote.Slice(0, 1).Span);
+            this.Flush();
+
+            this.binaryValueStream =  new ODataUtf8JsonWriteStream(this);
+
+            return this.binaryValueStream;
+        }
+
+        /// <summary>
+        /// Ends a scope for writing a stream value.
+        /// </summary>
+        public void EndStreamValueScope()
+        {
+            this.binaryValueStream.Dispose();
+            this.binaryValueStream = null;
+            this.Flush();
+
+            this.bufferWriter.Write(this.DoubleQuote.Slice(0, 1).Span);
+
+            CheckIfSeparatorNeeded();
+            CheckIfManualValueAtArrayStart();
+        }
+
+        /// <summary>
+        /// Asynchronously starts a scope for writing a stream value.
+        /// </summary>
+        /// <returns>A task representing the asynchronous operation. The task result contains a stream for writing the stream value.</returns>
+        public async Task<Stream> StartStreamValueScopeAsync()
+        {
+            this.WriteSeparatorIfNecessary();
+            this.bufferWriter.Write(this.DoubleQuote.Slice(0, 1).Span);
+            await this.FlushIfBufferThresholdReachedAsync().ConfigureAwait(false);
+
+            this.binaryValueStream = new ODataUtf8JsonWriteStream(this);
+
+            return this.binaryValueStream;
+        }
+
+        /// <summary>
+        /// Asynchronously ends a scope for writing a stream value.
+        /// </summary>
+        /// <returns>A task representing the asynchronous operation.</returns>
+        public async Task EndStreamValueScopeAsync()
+        {
+            await this.binaryValueStream.DisposeAsync().ConfigureAwait(false);
+            this.binaryValueStream = null;
+            await this.FlushIfBufferThresholdReachedAsync().ConfigureAwait(false);
+
+            this.bufferWriter.Write(this.DoubleQuote.Slice(0, 1).Span);
+
+            CheckIfSeparatorNeeded();
+            CheckIfManualValueAtArrayStart();
+        }
+
+        /// <summary>
+        /// Represents a stream for writing UTF-8 JSON data to an ODataUtf8JsonWriter.
+        /// </summary>
+        private sealed class ODataUtf8JsonWriteStream : Stream
+        {
+            private readonly ODataUtf8JsonWriter jsonWriter = null;
+            private byte[] buffer;
+            int numBytesNotWrittenFromPreviousChunk = 0;
+
+            /// <summary>
+            /// Initializes a new instance of the ODataUtf8JsonWriterStream class with the specified ODataUtf8JsonWriter.

```

Comment:
```suggestion
            /// Initializes a new instance of the ODataUtf8JsonWriteStream class with the specified ODataUtf8JsonWriter.
```
Or might be better to use `<see cref="ODataUtf8JsonWriteStream"/>`. That way the reference would be update in the event you renamed.